--- boost_1_45_0/boost/thread/detail/config.hpp	2010-07-10 03:13:09.000000000 +0800
+++ config.hpp	2011-01-10 01:12:06.339913800 +0800
@@ -37,7 +37,8 @@
 #elif defined(BOOST_THREAD_USE_LIB)   //Use lib
 #else //Use default
 #   if defined(BOOST_THREAD_PLATFORM_WIN32)
-#       if defined(BOOST_MSVC) || defined(BOOST_INTEL_WIN)
+#       if defined(BOOST_MSVC) || defined(BOOST_INTEL_WIN) \
+            || ( __GNUC__ >= 4 && __GNUC_MINOR__ >= 5 )
             //For compilers supporting auto-tss cleanup
             //with Boost.Threads lib, use Boost.Threads lib
 #           define BOOST_THREAD_USE_LIB
